test/framework/src/main/java/org/elasticsearch/node/MockNode.java

```diff
@@ -97,6 +100,21 @@ protected TransportService newTransportService(Settings settings, Transport tran
     }
 
     @Override
+    protected ZenPing newZenPing(Settings settings, ThreadPool threadPool, TransportService transportService,
+                                 UnicastHostsProvider hostsProvider) {
+        if (getPluginsService().filterPlugins(MockZenPing.TestPlugin.class).isEmpty()) {
```

Comment:
I followed the pattern we have used for other test only extensions. If we want to change to using test settings instead of marker interfaces that is fine, but we should do it as a separate change.